cdap-messaging-spi/src/main/java/io/cdap/cdap/messaging/MessagingService.java

```diff
@@ -39,7 +38,6 @@ public interface MessagingService {
    * @param topicMetadata topic to be created
    * @throws TopicAlreadyExistsException if the topic to be created already exist
    * @throws IOException if failed to create the topic
-   * @throws ServiceUnavailableException if the messaging service is not available
```

Comment:
Would it still do it for the LevelDB-based TMS? We would still have it for quite some time

Reply:
I just removed ServiceUnavailableException from comments as there is not direct reference to ServiceUnavailableException in the code. It'll be handled as before in case of TMS use.

Reply:
This guy can throw it `io.cdap.cdap.messaging.client.ClientMessagingService#handleError(int, java.util.function.Supplier<java.lang.String>, java.lang.String)`.
It's RuntimeException, so it's not required to declare it in throws, but JavaDoc helps caller to understand it can come out

Reply:
Done